test/support/utils.h

```diff
@@ -304,6 +304,18 @@ operator<(const MemoryChecker& v1, const MemoryChecker& v2)
     return v1.value() < v2.value();
 }
 
+template <typename T>
+struct SequenceData
+{
+    using DataType = T;
+};
+
+template <>
+struct SequenceData<bool>
+{
+    using DataType = unsigned char;
+};
+
```

Comment:
We claim we test `bool` at a test level, but we change it to `unsigned char` here for some not obvious (at least for me) reason. I believe such substitution should be visible at a test level; it might be even better to do replacement right in the tests. Could you describe the reason to use `unsigned char` instead of `bool`? Do we have some limitation in the library that should be documented?

Reply:
The root cause of this compile error is that std::vector<bool> is not a vector of some bools, it is a vector of bytes. Each item of source vector is only one bit in memory of this bytes vector: https://en.cppreference.com/w/cpp/container/vector_bool
So we can't use std::vector<bool> with our algorithms because of this limitation: SYCL library now sometimes tried to get address of vector item, but for std::vector<bool> this operation is not applicable.
@ValentinaKats please add this limitation to our doc.

Reply:
Dmitriy is asking correct questions. I don't remember the decision about making bool unavailable, users might have other containers rather than vector<bool>. If we think we do not allow to use that why we silently replace it by `unsigned char`? I am also not sure it's the right direction to fix that.

Reply:
This is only test environment. The STL library does the same for std::vector<bool>

Reply:
Fixed by other way: replace in tests Sequence<bool> by Sequence<unsigned char>

Reply:
Reverted